Theme_manager

mumok4 opened the pull request:
Добавил класс для работы с темами, для него написал json конфиг, в узлах заменил цвета на темы, в abstract_node заменил старую привязку на применение при помощи темового менеджера, в anot_node сделал применение тем к пинам

mumok4 committed:
#89, waiting review, added theme manager, replaced old theme binding

добавил класс для работы с темами, для него написал json конфиг, в узлах заменил цвета на темы, в abstract_node заменил старую привязку на применение при помощи темового менеджера, в anot_node сделал применение тем к пинам

jezvgg reviewed (changes requested):
Бля, ну слушай переделай немного и объединение тем добавь и хорошо будет

mumok4 committed:
#89, waiting review, added themes composition

Сделал так, чтобы apply принимал список тем, объединял их и применял, конечно сохранял в словарь, если есть несколько параметров одного аттрибута, то должен применяться последний

mumok4 committed:
#89, waiting review, added new methods to theme_manager

Добавлены добавление и убирание тем с объекта, методы: apply_theme перезаписывает темы объекта, add_theme добавляет темы к уже существующим, remove_theme убирает темы из списка тем объекта

mumok4 committed:
#35, waiting review, added tests, #89, waiting review, small args changes

Добавлены тесты для темового менеджера, также поменяны местами аргументы в некоторых методах темового менеджера, для единообразия

mumok4 committed:
#35, waiting review, added tests

jezvgg reviewed (changes requested):
Отлично получилось, нужно только добавить немного синтаксического сахара и использовать логические операции.

mumok4 committed:
#89, waiting review, refactored theme_manager, #35, waiting review, refactored test

Добалена распаковка значений списка в методах, там где можно, использованы логическик операторы, добавлен defaultdict, обновлены тесты

mumok4 committed:
#89, waiting review, small code refactoring, #35, waiting review, added ThemeManager to setup

Добавил |= и -=(^= - XOR, работал бы странно), немного изменил условия в create_theme

jezvgg reviewed (approved):
Крутяк
